client/console/src/main/java/org/apache/syncope/client/console/panels/HistoryConfDetails.java

```diff
@@ -0,0 +1,259 @@
+                    LOG.error("While updating console layout info for history configurations {}", infos, e);
+                    SyncopeConsoleSession.get().error(StringUtils.isBlank(e.getMessage())
+                            ? e.getClass().getName() : e.getMessage());
+                }
+                ((BasePage) pageRef.getPage()).getNotificationPanel().refresh(target);
+            }
+        };
+
+        replace(jsonEditorPanel);
+    }
+
+    private ConsoleLayoutInfo getJSONInfo(final T historyConfTO) {
+        Object conf = null; // current configuration instance
+        String key = "";
+        if (historyConfTO instanceof ConnInstanceHistoryConfTO) {
+            ConnInstanceHistoryConfTO historyConf = ConnInstanceHistoryConfTO.class.cast(historyConfTO);
+            conf = historyConf.getConnInstanceTO();
+            key = historyConf.getKey();
+        } else if (historyConfTO instanceof ResourceHistoryConfTO) {
+            ResourceHistoryConfTO historyConf = ResourceHistoryConfTO.class.cast(historyConfTO);
+            conf = historyConf.getResourceTO();
+            key = historyConf.getKey();
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        String json = "";
+        try {
+            json = mapper.writeValueAsString(conf);
+        } catch (IOException ex) {
+            DirectoryPanel.LOG.error("Error converting objects to JSON", ex);
+        }
+
+        ConsoleLayoutInfo info = new ConsoleLayoutInfo(key);
+        info.setContent(json);
+
+        return info;
+    }
+
+    private Map<String, String> getDropdownNamesMap(final List<T> historyConfTOs) {
+        Map<String, String> historyConfMap = new HashMap<>();
+        if (currentHistoryConfTO instanceof ConnInstanceHistoryConfTO) {
+            for (T historyConfValue : historyConfTOs) {
+                ConnInstanceHistoryConfTO historyConf = ConnInstanceHistoryConfTO.class.cast(historyConfValue);
+                historyConfMap.put(historyConf.getKey(),
+                        historyConf.getCreator() + " - " + SyncopeConsoleSession.get().getDateFormat().format(
+                        historyConf.getCreation()) + " - " + historyConf.getKey());
+            }
+        } else if (currentHistoryConfTO instanceof ResourceHistoryConfTO) {
+            for (T historyConfValue : historyConfTOs) {
+                ResourceHistoryConfTO historyConf = ResourceHistoryConfTO.class.cast(historyConfValue);
+                historyConfMap.put(historyConf.getKey(),
+                        historyConf.getCreator() + " - " + SyncopeConsoleSession.get().getDateFormat().format(
+                        historyConf.getCreation()) + " - " + historyConf.getKey());
+            }
+        }
+        return historyConfMap;
+    }
+
+    private Form initDropdownDiffConfForm() {
+        final Form form = new Form("form");
+        form.setModel(new CompoundPropertyModel(currentHistoryConfTO));
+        form.setOutputMarkupId(Boolean.TRUE);
```

Comment:
I don't see the point in using `Boolean` in such places. rather than primitive `true`.